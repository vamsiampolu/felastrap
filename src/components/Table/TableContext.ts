import { createContext } from "react";
import type { Variant } from "../common";

export type TableContextValue = {
  bordered: boolean;
  borderless: boolean;
  variant: Variant;
  striped: boolean;
  hover: boolean;
};

export const TableContext = createContext<TableContextValue>({
  bordered: false,
  borderless: false,
  variant: "default",
  striped: false,
  hover: false,
});
