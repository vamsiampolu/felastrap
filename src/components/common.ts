import type * as Css from "csstype";
import type * as Fela from "fela";
import type {
  ComponentPropsWithoutRef,
  ComponentType,
  ElementType,
} from "react";
import type * as ReactFela from "react-fela";

export type ExcludeFrom<T, U> = T extends U ? never : T;
export type Theme = object;

export type Rule = Fela.TRule;
export type CSSObject = Fela.CSSObject;
export type StyleProps<P = object> = ReactFela.StyleProps<Theme, P>;
export type StyleFunction<P = object> = ReactFela.StyleFunction<Theme, P>;

export type Padding = Required<
  Pick<
    Css.Properties,
    "paddingTop" | "paddingLeft" | "paddingRight" | "paddingBottom"
  >
>;
export type BorderRadius = Required<
  Pick<
    Css.Properties,
    | "borderTopLeftRadius"
    | "borderTopRightRadius"
    | "borderBottomRightRadius"
    | "borderBottomLeftRadius"
  >
>;

export type PaddingString = Required<Css.Properties["padding"]>;
export type Color = Css.Property.Color;

export type HeaderTypes = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
export type Size = "small" | "medium" | "large";
export type Variant =
  | "default"
  | "danger"
  | "info"
  | "primary"
  | "secondary"
  | "success"
  | "warning";

export type FontSize = Css.Property.FontSize;
export type ColorMapping = Record<Variant, Color>;
export type PaddingBySize = Record<Size, Padding>;

export type SpanProps = ComponentPropsWithoutRef<"span">;
export type DivProps = ComponentPropsWithoutRef<"div">;
export type ListItemProps = ComponentPropsWithoutRef<"li">;

export type DropdownDirection = "up" | "down" | "end" | "start" | "centered";
export type CaretDirection = "up" | "down" | "end" | "start";
export type NavVariant =
  | "default"
  | "tabs"
  | "pills"
  | "underline"
  | "vertical";

export type CaretProps = DivProps & {
  width: string | number | null;
};

export type DynamicElement<T extends ElementType = "span"> =
  | keyof JSX.IntrinsicElements
  | ComponentType<T>;
export type PargraphProps = ComponentPropsWithoutRef<"p">;
export type ImageProps = ComponentPropsWithoutRef<"img">;
export type TooltipPlacement = "top" | "left" | "bottom" | "right";

export type MediaQuerySize = "xs" | "sm" | "md" | "lg" | "xl" | "xxl";
export type Timer = ReturnType<typeof setTimeout>;
export type OnClick<T extends ElementType> =
  ComponentPropsWithoutRef<T>["onClick"];
export type StyleKey = Exclude<
  keyof HTMLElement["style"],
  "length" | "parentRule"
>;
