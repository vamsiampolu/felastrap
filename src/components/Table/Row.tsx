import { ComponentPropsWithoutRef, useContext } from "react";
import { useFela } from "react-fela";
import { TableContext } from "./TableContext";
import { variantRule } from "./variant";
import { StyleProps } from "../common";

const rowRule = () => ({
  padding: "0.5rem 0.5rem",
  color: "initial",
  backgroundColor: "#fff",
  borderBottomWidth: "1px",
  boxShadow: "inset 0 0 0 9999px initial",
});

const borderedRule = ({ bordered }: StyleProps<{ bordered: boolean }>) =>
  bordered
    ? {
        borderWidth: "0 1px",
      }
    : {};

const borderlessRule = ({ borderless = false }) =>
  borderless
    ? {
        borderBottomWidth: 0,
      }
    : {};

export type TableRowProps = ComponentPropsWithoutRef<"tr"> & {
  oddRow?: boolean;
  header?: boolean;
  active?: boolean;
};

export const TableRow = ({
  children,
  className = "",
  header = false,
  ...props
}: TableRowProps) => {
  const {
    variant = "default",
    bordered = false,
    borderless = false,
  } = useContext(TableContext);
  const { css } = useFela({
    variant,
    bordered,
    borderless,
    header,
  });
  const cx = `${css([
    rowRule,
    variantRule,
    borderedRule,
    borderlessRule,
  ])} table-row ${className}`.trim();

  return (
    <tr data-testid="table-row" className={cx} {...props}>
      {children}
    </tr>
  );
};
