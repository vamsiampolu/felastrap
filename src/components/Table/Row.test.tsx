import { IRenderer } from "fela";
import { TableRow } from "./Row";
import { TableContext, TableContextValue } from "./TableContext";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../test-helpers";
import { render, screen } from "@testing-library/react";
import { Variant } from "../common";

describe("components/Table/Row", () => {
  let renderer: IRenderer;

  beforeAll(() => {
    renderer = createRenderer();
  });

  beforeEach(() => {
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a Table Row with the default context", async () => {
    const component = (
      <table>
        <TableContext.Provider value={{} as TableContextValue}>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <tbody>
            <TableRow>
              <td>Value A</td>
              <td>Value B</td>
            </TableRow>
          </tbody>
        </TableContext.Provider>
      </table>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("table-row")).toBeInTheDocument();

    expect(screen.getByTestId("table-row")).toHaveClass("table-row");
    expect(screen.getByTestId("table-row")?.tagName).toBe("TR");

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it("should render a Table Row with the context", async () => {
    const value: TableContextValue = {
      variant: "default",
      bordered: false,
      borderless: false,
      striped: false,
      hover: false,
    };

    const component = (
      <table>
        <TableContext.Provider value={value}>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <tbody>
            <TableRow>
              <td>Value A</td>
              <td>Value B</td>
            </TableRow>
          </tbody>
        </TableContext.Provider>
      </table>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);
    expect(screen.getByTestId("table-row")).toBeInTheDocument();

    expect(screen.getByTestId("table-row")).toHaveClass("table-row");
    expect(screen.getByTestId("table-row")?.tagName).toBe("TR");

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it.each([
    "default",
    "danger",
    "info",
    "primary",
    "secondary",
    "success",
    "warning",
  ])("should render a table row of variant %s", async (variant) => {
    const value: TableContextValue = {
      variant: variant as Variant,
      bordered: false,
      borderless: false,
      striped: false,
      hover: false,
    };

    const component = (
      <table>
        <TableContext.Provider value={value}>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <tbody>
            <TableRow>
              <td>Value A</td>
              <td>Value B</td>
            </TableRow>
          </tbody>
        </TableContext.Provider>
      </table>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("table-row")).toBeInTheDocument();
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it.each([
    "default",
    "danger",
    "info",
    "primary",
    "secondary",
    "success",
    "warning",
  ])("should render a bordered table row of variant %s", async (variant) => {
    const value: TableContextValue = {
      variant: variant as Variant,
      bordered: true,
      borderless: false,
      striped: false,
      hover: false,
    };

    const component = (
      <table>
        <TableContext.Provider value={value}>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <tbody>
            <TableRow>
              <td>Value A</td>
              <td>Value B</td>
            </TableRow>
          </tbody>
        </TableContext.Provider>
      </table>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("table-row")).toBeInTheDocument();
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });

  it.each([
    "default",
    "danger",
    "info",
    "primary",
    "secondary",
    "success",
    "warning",
  ])("should render a borderless table row of variant %s", async (variant) => {
    const value: TableContextValue = {
      variant: variant as Variant,
      bordered: false,
      borderless: true,
      striped: false,
      hover: false,
    };

    const component = (
      <table>
        <TableContext.Provider value={value}>
          <thead>
            <tr>
              <td>Column A</td>
              <td>Column B</td>
            </tr>
          </thead>
          <tbody>
            <TableRow>
              <td>Value A</td>
              <td>Value B</td>
            </TableRow>
          </tbody>
        </TableContext.Provider>
      </table>
    );

    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("table-row")).toBeInTheDocument();
    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
