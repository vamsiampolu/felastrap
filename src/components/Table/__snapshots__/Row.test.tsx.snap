// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Table/Row should render a Table Row with the context 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #dee2e6;
}
.f {
  border-right-color: #dee2e6;
}
.g {
  border-bottom-color: #dee2e6;
}
.h {
  border-left-color: #dee2e6;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a Table Row with the default context 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #dee2e6;
}
.f {
  border-right-color: #dee2e6;
}
.g {
  border-bottom-color: #dee2e6;
}
.h {
  border-left-color: #dee2e6;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant danger 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #f8d7da;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #c6acae;
}
.i {
  border-right-color: #c6acae;
}
.j {
  border-bottom-color: #c6acae;
}
.k {
  border-left-color: #c6acae;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant default 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #dee2e6;
}
.i {
  border-right-color: #dee2e6;
}
.j {
  border-bottom-color: #dee2e6;
}
.k {
  border-left-color: #dee2e6;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant info 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #cff4fc;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #a6c3ca;
}
.i {
  border-right-color: #a6c3ca;
}
.j {
  border-bottom-color: #a6c3ca;
}
.k {
  border-left-color: #a6c3ca;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant primary 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #cfe2ff;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #a6b5cc;
}
.i {
  border-right-color: #a6b5cc;
}
.j {
  border-bottom-color: #a6b5cc;
}
.k {
  border-left-color: #a6b5cc;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant secondary 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #e2e3e5;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #b5b6b7;
}
.i {
  border-right-color: #b5b6b7;
}
.j {
  border-bottom-color: #b5b6b7;
}
.k {
  border-left-color: #b5b6b7;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant success 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #d1e7dd;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #a7b9b1;
}
.i {
  border-right-color: #a7b9b1;
}
.j {
  border-bottom-color: #a7b9b1;
}
.k {
  border-left-color: #a7b9b1;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a bordered table row of variant warning 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff3cd;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-width: 0;
}
.f {
  border-right-width: 1px;
}
.g {
  border-left-width: 1px;
}
.h {
  border-top-color: #ccc2a4;
}
.i {
  border-right-color: #ccc2a4;
}
.j {
  border-bottom-color: #ccc2a4;
}
.k {
  border-left-color: #ccc2a4;
}
.l {
  padding-top: 0.5rem;
}
.m {
  padding-right: 0.5rem;
}
.n {
  padding-bottom: 0.5rem;
}
.o {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l m n o table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant danger 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #f8d7da;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #c6acae;
}
.f {
  border-right-color: #c6acae;
}
.g {
  border-bottom-color: #c6acae;
}
.h {
  border-left-color: #c6acae;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant default 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #dee2e6;
}
.f {
  border-right-color: #dee2e6;
}
.g {
  border-bottom-color: #dee2e6;
}
.h {
  border-left-color: #dee2e6;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant info 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #cff4fc;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #a6c3ca;
}
.f {
  border-right-color: #a6c3ca;
}
.g {
  border-bottom-color: #a6c3ca;
}
.h {
  border-left-color: #a6c3ca;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant primary 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #cfe2ff;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #a6b5cc;
}
.f {
  border-right-color: #a6b5cc;
}
.g {
  border-bottom-color: #a6b5cc;
}
.h {
  border-left-color: #a6b5cc;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant secondary 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #e2e3e5;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #b5b6b7;
}
.f {
  border-right-color: #b5b6b7;
}
.g {
  border-bottom-color: #b5b6b7;
}
.h {
  border-left-color: #b5b6b7;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant success 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #d1e7dd;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #a7b9b1;
}
.f {
  border-right-color: #a7b9b1;
}
.g {
  border-bottom-color: #a7b9b1;
}
.h {
  border-left-color: #a7b9b1;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a borderless table row of variant warning 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff3cd;
}
.c {
  border-bottom-width: 0;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #ccc2a4;
}
.f {
  border-right-color: #ccc2a4;
}
.g {
  border-bottom-color: #ccc2a4;
}
.h {
  border-left-color: #ccc2a4;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant danger 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #f8d7da;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #c6acae;
}
.f {
  border-right-color: #c6acae;
}
.g {
  border-bottom-color: #c6acae;
}
.h {
  border-left-color: #c6acae;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant default 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #dee2e6;
}
.f {
  border-right-color: #dee2e6;
}
.g {
  border-bottom-color: #dee2e6;
}
.h {
  border-left-color: #dee2e6;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant info 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #cff4fc;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #a6c3ca;
}
.f {
  border-right-color: #a6c3ca;
}
.g {
  border-bottom-color: #a6c3ca;
}
.h {
  border-left-color: #a6c3ca;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant primary 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #cfe2ff;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #a6b5cc;
}
.f {
  border-right-color: #a6b5cc;
}
.g {
  border-bottom-color: #a6b5cc;
}
.h {
  border-left-color: #a6b5cc;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant secondary 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #e2e3e5;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #b5b6b7;
}
.f {
  border-right-color: #b5b6b7;
}
.g {
  border-bottom-color: #b5b6b7;
}
.h {
  border-left-color: #b5b6b7;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant success 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #d1e7dd;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #a7b9b1;
}
.f {
  border-right-color: #a7b9b1;
}
.g {
  border-bottom-color: #a7b9b1;
}
.h {
  border-left-color: #a7b9b1;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;

exports[`components/Table/Row should render a table row of variant warning 1`] = `
".a {
  color: #000;
}
.b {
  background-color: #fff3cd;
}
.c {
  border-bottom-width: 1px;
}
.d {
  box-shadow: inset 0 0 0 9999px initial;
}
.e {
  border-top-color: #ccc2a4;
}
.f {
  border-right-color: #ccc2a4;
}
.g {
  border-bottom-color: #ccc2a4;
}
.h {
  border-left-color: #ccc2a4;
}
.i {
  padding-top: 0.5rem;
}
.j {
  padding-right: 0.5rem;
}
.k {
  padding-bottom: 0.5rem;
}
.l {
  padding-left: 0.5rem;
}


<table>
  <thead>
    <tr>
      <td>Column A</td>
      <td>Column B</td>
    </tr>
  </thead>
  <tbody>
    <tr data-testid="table-row" class="a b c d e f g h i j k l table-row">
      <td>Value A</td>
      <td>Value B</td>
    </tr>
  </tbody>
</table>
;
"
`;
